cmake/bintools/gnu/target.cmake

```diff
@@ -11,7 +11,7 @@ find_program(CMAKE_READELF ${CROSS_COMPILE}readelf PATHS ${TOOLCHAIN_HOME} NO_DE
 find_program(CMAKE_NM      ${CROSS_COMPILE}nm      PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 find_program(CMAKE_STRIP   ${CROSS_COMPILE}strip   PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 
-find_program(CMAKE_GDB     ${CROSS_COMPILE}gdb     PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+find_program(CMAKE_GDB     ${CROSS_COMPILE}gdb-py  PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
```

Comment:
this code is not only Zephyr SDK specific code, so there could easily be 3rd party toolchains that we support which ships `gdb` only (detected below, I know).

I recalled we did discuss the `gdb` vs `gdb-py` vs `gdb-no-py` and I do see 3rd party toolchains shipping `gdb-py` as the python version, and `gdb` as the non-py.
Not seeing any `gdb-no-py` so I can't recall from earlier discussion if this is standard naming convention for non-python gdb, or something we chose in Zephyr SDK.

Due to this, we could consider this test instead:
```
# Prefer `gdb-py` if found, else fallback on `gdb`
find_program(CMAKE_GDB ${CROSS_COMPILE}gdb-py ${CROSS_COMPILE}gdb PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
```

then test the gdb we found, and if that gdb fails, then do the second lookup as:
```
find_program(CMAKE_GDB_NO_PY ${CROSS_COMPILE}gdb-no-py ${CROSS_COMPILE}gdb PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
```
This means we will try any `gdb-py` version before the `gdb`, but still test a `gdb` if found in first lookup.

and in case where no `gdb-py` is found and `gdb` is found and fails the test, then gdb-no-py` is having highest priority in second lookup.

Reply:
`gdb-no-py` is a Zephyr SDK specific thing (prior to 0.15.0, that is). Since the minimum required Zephyr SDK has been updated to 0.15.0, there is no need to keep it around.

Reply:
> I do see 3rd party toolchains shipping `gdb-py` as the python version, and `gdb` as the non-py.

The only other toolchain that does, besides the Zephyr SDK 0.15.0, is GNU Arm Embedded:
```
$ ls /opt/gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gdb*
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gdb
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gdb-add-index
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gdb-add-index-py
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/arm-none-eabi-gdb-py
```